core/src/main/java/org/apache/accumulo/core/cli/ClientOpts.java

```diff
@@ -161,6 +155,7 @@ public void setPrincipal(String principal) {
   }
 
   public void setClientProperties(Properties clientProps) {
+    ClientProperty.validate(clientProps);
     this.cachedProps = clientProps;
```

Comment:
What if the principal is set?

Reply:
I am going to do another PR to remove these setters